src/mongocxx/options/find.hpp

```diff
@@ -455,6 +455,16 @@ class MONGOCXX_API find {
     ///
     const stdx::optional<bsoncxx::document::view_or_value>& sort() const;
 
+    ///
+    /// Modifies the object to convert all modifiers into their respective top-level options. For
+    /// internal use only.
+    ///
+    /// @throws
+    ///   mongocxx::exception if any modifiers are invalid, or if the unsupported "$query" or
+    ///   "$explain" modifiers are set.
+    ///
+    MONGOCXX_PRIVATE find& convert_all_modifiers();
```

Comment:
Happy to receive opinions on both of the following:
1. Should this mutate an options::find object, or should it return a copy of the options::find object it's invoked on?
2. Should this be a public non-static class method, or a public static class method, or a free function?  It doesn't particularly need access to any internal state.